### ISSUE TITLE
Issue with testing-library/render-result-naming-convention 

### ISSUE THREAD

**Issue by rdelgatte** [testing-library/eslint-plugin-testing-library#379]
Since migrating to the latest version of the library, I have some troubles / questions about this rule as it triggers an error whenever I have a function whose name contains `render` even though it isn't related to `testing-library` at all.

Here is the dumbest example I have:
![image](https://user-images.githubusercontent.com/11570518/117957777-ae510900-b31a-11eb-9360-22e54af56c87.png)

```ts
const renderAnything = () => 'anything';
const result = renderAnything();
```

It even fails with some other function names like `surrender`, `rerender`,... you got my point I guess :-) 

Renaming the function to avoid this `render` keyword fix the issue but I think it should not raise any issue if it isn't related to `testing-library` IMO.

**Comment by Belco90** [comment]
Hey @rdelgatte! This is probably related to Aggressive Reporting. Have you checked [the Troubleshooting section](https://github.com/testing-library/eslint-plugin-testing-library#troubleshooting) to find more about restricting this behavior?

**Comment by rdelgatte** [comment]
Hello @Belco90 I've tried digging into this section (thank you for the link, I didn't see it) but I can't figure out how to deal with that. 
Does this apply for all the rules (as `testing-library/render-result-naming-convention`)?
What if I want the examples below to not raise any issue because it is not related to testing-library `render` function? 

I wouldn't expect the result variable to be either `view` or `utils` as it does not actually use the library.

Thank you in advance for your help!

**Comment by MichaelDeBoey** [comment]
@rdelgatte Do you maybe have a reproduction repo where you have the least amount of code to come across this error?
That could help us in understanding your use-case.

**Comment by Belco90** [comment]
> Does this apply for all the rules (as `testing-library/render-result-naming-convention`)?
Yes. The new detection mechanism introduced in version 4 applies to all rules. Depending on what is being reported, then different parts of the mechanism related to imports, renders, or queries would apply.

> What if I want the examples below to not raise any issue because it is not related to testing-library `render` function?
> 
> I wouldn't expect the result variable to be either `view` or `utils` as it does not actually use the library.

It depends on how you want to narrow `eslint-plugin-testing-library` reporting! If you are getting those errors reported in non-testing files, then [set up the plugin to be only run against your testing files](https://github.com/testing-library/eslint-plugin-testing-library#run-the-plugin-only-against-test-files) would be enough.

If you are getting those errors in testing files, then you'd to configure some [shared settings](https://github.com/testing-library/eslint-plugin-testing-library#shared-settings). By default, `eslint-plugin-testing-library` will assume all utils matching a certain pattern must be reported, no matter where they are imported from. If you want `eslint-plugin-testing-library` to check from where the utils are imported from, you can use [`testing-library/utils-module` shared setting](https://github.com/testing-library/eslint-plugin-testing-library#testing-libraryutils-module). If have a custom module from where you re-export Testing Library utils, you need to set that in this setting so only things imported from `@testing-library/*` or the indicated module would be reported. If you don't have any custom module, you can just switch the setting off setting it to `"off"` so only things imported from `@testing-library/*` are reported.

Another possibility if you are getting errors in testing files is to restrict the list of methods that are considered `render`. By default, `eslint-plugin-testing-library` will assume any function call named `*render*` is a Testing Library `render`. For opting this out you can use [`custom-renders` shared setting](https://github.com/testing-library/eslint-plugin-testing-library#testing-librarycustom-renders). By setting this option, the only function calls strictly named `render` or as one of the indicated options will be reported.

You can even combine all these settings, but I don't think it's necessary. I hope that helps!
